ngraph/core/src/op/experimental_detectron_topkrois.cpp

```diff
@@ -0,0 +1,84 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/experimental_detectron_topkrois.hpp"
+#include "itt.hpp"
+#include "ngraph/attribute_visitor.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "ngraph/validation_util.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::v6::ExperimentalDetectronTopKROIs, "ExperimentalDetectronTopKROIs", 6);
+
+op::v6::ExperimentalDetectronTopKROIs::ExperimentalDetectronTopKROIs(const Output<Node>& input_rois,
+                                                                     const Output<Node>& rois_probs,
+                                                                     size_t max_rois)
+    : Op({input_rois, rois_probs})
+    , m_max_rois(max_rois)
+{
+    constructor_validate_and_infer_types();
+}
+
+bool op::v6::ExperimentalDetectronTopKROIs::visit_attributes(AttributeVisitor& visitor)
+{
+    NGRAPH_OP_SCOPE(v6_ExperimentalDetectronTopKROIs_visit_attributes);
+    visitor.on_attribute("max_rois", m_max_rois);
+    return true;
+}
+
+shared_ptr<Node>
+    op::v6::ExperimentalDetectronTopKROIs::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    NGRAPH_OP_SCOPE(v6_ExperimentalDetectronTopKROIs_clone_with_new_inputs);
+    check_new_args_count(this, new_args);
+    return make_shared<op::v6::ExperimentalDetectronTopKROIs>(
+        new_args.at(0), new_args.at(1), m_max_rois);
+}
+
+void op::v6::ExperimentalDetectronTopKROIs::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v6_ExperimentalDetectronTopKROIs_validate_and_infer_types);
+    const auto input_rois_shape = get_input_partial_shape(0);
+    const auto rois_probs_shape = get_input_partial_shape(1);
+
+    if (input_rois_shape.rank().is_static() && rois_probs_shape.rank().is_static())
```

Comment:
Let's make the checks more specific and get rid of one big `if`. If just one of the inputs is static (for example, `input_rois_shape`) its additional values should be checked.

Reply:
Done.